torch/_prims/__init__.py

```diff
@@ -0,0 +1,992 @@
+    impl_aten=torch.erf,
+    doc="",
+)
+
+erf_inv = _make_elementwise_unary_prim(
+    impl_aten=torch.special.erfinv,
+    doc="",
+)
+
+erfc = _make_elementwise_unary_prim(
+    impl_aten=torch.special.erfc,
+    doc="",
+)
+
+exp = _make_elementwise_unary_prim(
+    impl_aten=torch.exp,
+    doc="",
+)
+
+expm1 = _make_elementwise_unary_prim(
+    impl_aten=torch.special.expm1,
+    doc="",
+)
+
+floor = _make_elementwise_unary_prim(
+    impl_aten=torch.floor,
+    doc="",
+)
+
+igamma = _make_elementwise_unary_prim(
+    impl_aten=torch.special.gammainc,
+    doc="",
+)
+
+igammac = _make_elementwise_unary_prim(
+    impl_aten=torch.special.gammaincc,
+    doc="",
+)
+
+is_finite = _make_elementwise_unary_prim(
+    impl_aten=torch.isfinite,
+    doc="",
+)
+
+lgamma = _make_elementwise_unary_prim(
+    impl_aten=torch.lgamma,
+    doc="",
+)
+
+log = _make_elementwise_unary_prim(
+    impl_aten=torch.log,
+    doc="",
+)
+
+log1p = _make_elementwise_unary_prim(
+    impl_aten=torch.log1p,
+    doc="",
+)
+
+reciprocal = _make_elementwise_unary_prim(
+    impl_aten=torch.reciprocal,
+    doc="",
+)
+
+neg = _make_elementwise_unary_prim(
+    impl_aten=torch.neg,
+    doc="",
+)
+
+round = _make_elementwise_unary_prim(
+    impl_aten=torch.round,
+    doc="",
+)
+
+sign = _make_elementwise_unary_prim(
+    impl_aten=torch.sign,
+    doc="",
+)
+
+sin = _make_elementwise_unary_prim(
+    impl_aten=torch.sin,
+    doc="",
+)
+
+sinh = _make_elementwise_unary_prim(
+    impl_aten=torch.sinh,
+    doc="",
+)
+
+sqrt = _make_elementwise_unary_prim(
+    impl_aten=torch.sqrt,
+    doc="",
+)
+
+square = _make_elementwise_unary_prim(
+    impl_aten=torch.square,
+    doc="",
+)
+
+tan = _make_elementwise_unary_prim(
+    impl_aten=torch.tan,
+    doc="",
+)
+
+#
+# Elementwise binary operations
+#
+
+add = _make_elementwise_binary_prim(
+    impl_aten=torch.add,
+    doc="",
+)
+
+atan2 = _make_elementwise_binary_prim(
+    impl_aten=torch.atan2,
+    doc="",
+)
+
+bitwise_and = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_and,
+    doc="",
+)
+
+bitwise_not = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_not,
+    doc="",
+)
+
+bitwise_or = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_or,
+    doc="",
+)
+
+bitwise_xor = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_xor,
+    doc="",
+)
+
+# TODO: complex needs a special meta to account for its float -> complex behavior
+# complex = _make_elementwise_binary_prim(
+#   impl_aten=torch.complex,
+#   doc="",
+# )
+
+# div prim performs truncation division on integer inputs
+#   and true division for floating and complex inputs
+def _div(a, b):
+    if isinstance(a, (bool, int)):
+        return torch.div(a, b, rounding_mode="trunc")
+    return torch.true_divide(a, b)
+
+
+div = _make_elementwise_binary_prim(
+    impl_aten=_div,
+    doc="",
+)
+
+eq = _make_elementwise_binary_prim(
+    impl_aten=torch.eq,
+    doc="",
+)
```

Comment:
this look ill-typed; eq is supposed to return a bool tensor right

Reply:
Yes eq returns a bool tensor, why?

Reply:
`_make_elementwise_binary_prim` means it uses `_elementwise_meta`  which is dtype preserving, but you instead want to force the dtype to bool

Reply:
Oh you mean the meta is wrong -- yes most meta functions are not tested yet